hash_practice/exercises.py

```diff
@@ -1,19 +1,35 @@
-
+import collections
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(N log N)

```

Comment:
My guess is that you were attempting to account for the call to `sorted` on line 13. But be careful of mixing your variables. The call to `sorted` _is_ O(n * log(n)), but only if `n` refers to the number of characters in each string. But on top of that, `sorted` is called for each string in the list.

So let's use `n` to be the number of strings in the list and `m` to be the number characters in each string. That would give us O(n * m * log(m)). However, if we make the simplifying assumption that each string in the list is an English word, then we don't need to worry about `m` so much. English words tend not to get so long, about 4.7 letters on average. That is going to be vastly dwarfed by the number of strings in the list (`n`), which could easily be hundreds or thousands of strings. In that case, we can ignore `m` and say O(n).

Your calculation for the space complexity (assuming `n` is the number of strings in the list) and implementation are fine, though.